yew-functional/src/lib.rs

```diff
@@ -173,204 +169,76 @@ where
     }
 }
 
-pub fn use_ref<T: 'static, InitialProvider>(initial_value: InitialProvider) -> Rc<RefCell<T>>
-where
-    InitialProvider: FnOnce() -> T,
-{
-    #[derive(Clone)]
-    struct UseRefState<T>(Rc<RefCell<T>>);
-    impl<T> Hook for UseRefState<T> {}
-
-    use_hook(
-        |state: &mut UseRefState<T>, hook_callback| {
-            // we need it to be a specific closure type, even if we never use it
-            let _ignored = || hook_callback(|_| false, false);
-            state.0.clone()
-        },
-        move || UseRefState(Rc::new(RefCell::new(initial_value()))),
-    )
+pub fn get_current_scope() -> Option<AnyScope> {
```

Comment:
The visibility changed from `pub(crate)` to `pub`, what's the reasoning here?

Reply:
I've found it useful to expose the scope + context without subscribing to downstream updates - which is what the current use_context does. React has the same problem - anyone using use_context now needs to re-run on every update to the context, even if those updates wouldn't cause an update in the component. This is the reason why Recoil was made - there are no use_context selectors because the primitives don't expose such a mechanism. Recoil sidesteps context a bit. Perhaps we could provide a static version of use_context that suppresses updates to its children? Or have a way of configuring use_context?

Open to ideas.

```rust
        let scope = get_current_scope().expect(
            "No current Scope. `use_context` can only be called inside function components",
        );
        let recoil_scope =
            find_context_provider_scope(&scope).expect("Unable to find RecoilRoot in scope");
```

Reply:
> anyone using use_context now needs to re-run on every update to the context, even if those updates wouldn't cause an update in the component

I think this can still be circumvented with memoized hooks if the optimization is needed. Do you mean reverting this change and opening a new issue for discussion?

Reply:
Actually I'm in favor of this change, it allows for more advanced hooks to interact with parent components without having to build on top of `use_context`